client/telemagent.go

```diff
@@ -30,4 +39,80 @@ func (c *Client) DeviceSession() (deviceSession []string, err error) {
 	}
 
 	return
+}
+
+// Login logs user in.
+func (client *Client) CheckEmail(email, password, otp string) (bool, error) {
+	postData := loginData{
+		Email:    email,
+		Password: password,
+		Otp:      otp,
+	}
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(postData); err != nil {
+		return false, err
+	}
+
+	var user User
+	if _, err := client.doSync("POST", "/v2/login", nil, nil, &body, &user); err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (c *Client) Associate(email string, password string, otp string) error {
+	url := os.Getenv("TELEMGW_SERVICE_URL")
+
+	isEmailOk, err := c.CheckEmail(email, password, otp)
```

Comment:
Why not use `client.WhoAmI()`?

That way, they just do `snap login` first and then this. Granted this is a fork that we'll throw away it's not a big deal.. but if a user is already logged in, we can just use that instead